arangod/RestServer/DatabaseFeature.cpp

```diff
@@ -829,6 +829,19 @@ ErrorCode DatabaseFeature::dropDatabase(std::string_view name) {
     bool result = vocbase->markAsDropped();
     TRI_ASSERT(result);
 
+    // mark all collections in this database as deleted, too.
+    try {
+      auto collections = vocbase->collections(/*includeDeleted*/ false);
+      for (auto& c : collections) {
+        c->setDeleted();
+        c->deferDropCollection(TRI_vocbase_t::dropCollectionCallback);
+      }
+    } catch (std::exception const& ex) {
+      LOG_TOPIC("1f096", WARN, Logger::FIXME)
+          << "error while dropping collections in database '" << vocbase->name()
+          << "': " << ex.what();
+    }
```

Comment:
Regarding CPU intensiveness:
I don't think it matters a lot, because dropping an entire database is an operation that is not carried out often (if at all).
But freeing the resources of the collections from that database is really important, because otherwise they continue to hog memory forever.

Reply:
Regarding doing it in `markAsDropped()`: I don't think that helps, because markAsDropped only flips the internal status and does nothing else. It is also `noexcept` right now.
I checked that `markAsDropped()` doesn't have any more callers, so doing it inside the DatabaseFeature is not a problem. The DatabaseFeature also does several other things related to the database when it gets dropped, e.g. garbage-collecting stuff and invalidating caches. So I think marking all collections of the DB as dropped is not problematic to add here, too.

Reply:
Well, maybe my point wasn't clear. I don't have something against new sync behavior.
My main question was "do we need gc-like behavior at all in such case?"